molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/request/QueryGenerator.java

```diff
@@ -888,18 +888,31 @@ private Object getQueryValue(Attribute attribute, Object queryRuleValue)
 			case XREF:
 				return queryRuleValue instanceof Entity ? ((Entity) queryRuleValue).getIdValue() : queryRuleValue;
 			case DATE:
+				if (queryRuleValue instanceof LocalDate)
+				{
+					return queryRuleValue.toString();
+				}
+				else if (queryRuleValue instanceof String)
+				{
+					return queryRuleValue;
+				}
+				else
+				{
+					throw new MolgenisQueryException(format("Query value must be of type LocalDate instead of [%s]",
+							queryRuleValue.getClass().getSimpleName()));
+				}
 			case DATE_TIME:
-				if (queryRuleValue instanceof Date)
+				if (queryRuleValue instanceof Instant)
 				{
-					return getESDateQueryValue((Date) queryRuleValue, attribute);
+					return queryRuleValue.toString();
 				}
 				else if (queryRuleValue instanceof String)
 				{
 					return queryRuleValue;
 				}
 				else
 				{
-					throw new MolgenisQueryException(format("Query value must be of type Date instead of [%s]",
+					throw new MolgenisQueryException(format("Query value must be of type Instant instead of [%s]",
 							queryRuleValue.getClass().getSimpleName()));
```

Comment:
Will error also be posted to end user? Because then I think 'Instant' is not very informational. Instead provide info on how Instant should be formatted?